Packages/Tracking/Core/Runtime/Plugins/LeapCSharp/Connection.cs

```diff
@@ -1024,6 +1024,56 @@ static public eLeapPolicyFlag FlagForPolicy(Controller.PolicyFlag singlePolicy)
             }
         }
 
+        /// <summary>
+        /// Temporarily makes a connection to determine if a Service is available.
+        /// Returns the result and closes the temporary connection upon completion.
+        /// </summary>
+        public static bool IsConnectionAvailable(string serverNamespace = "Leap Service")
+        {
+            LEAP_CONNECTION_CONFIG config = new LEAP_CONNECTION_CONFIG();
+            config.server_namespace = Marshal.StringToHGlobalAnsi(serverNamespace);
+            config.flags = 0;
+            config.size = (uint)Marshal.SizeOf(config);
+
+            IntPtr tempConnection;
+
+            eLeapRS result;
+
+            result = LeapC.CreateConnection(ref config, out tempConnection);
+
+            if (result != eLeapRS.eLeapRS_Success || tempConnection == IntPtr.Zero)
+            {
+                LeapC.CloseConnection(tempConnection);
```

Comment:
If creating a connection fails, is there one to close?

Reply:
I think you may be right here, at _least_ with an IntPtr.Zero situation

Reply:
Though, CloseConnection will itself handle any pointer it is given, so it may be considered safer and neater to keep these closes in, rather than switch through all eventualities

Reply:
Yep, sounds OK if CloseConnection is defensively coded. I'm happy with that.